core/src/main/java/com/csi4999/systems/networking/serverlisteners/SaveCreatureListener.java

```diff
@@ -0,0 +1,102 @@
+package com.csi4999.systems.networking.serverlisteners;
+
+import com.csi4999.systems.creature.Creature;
+import com.csi4999.systems.networking.Database;
+import com.csi4999.systems.networking.SerializedType;
+import com.csi4999.systems.networking.packets.SaveCreaturePacket;
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryonet.Connection;
+import com.esotericsoftware.kryonet.Listener;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+/**
+ * Handles saving a creature on creation or update. Does not handle updating a creature's win-loss
+ * See tournament result packet (todo)
+ */
+public class SaveCreatureListener implements Listener {
+
+    private Kryo k;
+    private Database db;
+
+    private final String saveCreatureQuery = "INSERT INTO creature (wins,losses,user_id) VALUES(?,?,?);";
+    private final String creatureListQuery = "SELECT * FROM creature;";
+
+    public SaveCreatureListener(Database db, Kryo k) {
+        this.db = db;
+        this.k = k;
+    }
+
+    @Override
+    public void received(Connection c, Object o) {
+
+        if (o instanceof SaveCreaturePacket) {
+
+            System.out.println("Received Save Creature packet");
+            SaveCreaturePacket p = (SaveCreaturePacket) o;
+            trySave(c,p);
+        }
+
+    }
+
+    private void trySave(Connection c, SaveCreaturePacket p) {
```

Comment:
"Connection c" isn't used as a parameter here and can be removed

Reply:
fixed